website/pages/docs/k8s/crds/upgrade-to-crds.mdx

```diff
@@ -0,0 +1,318 @@
+---
+layout: docs
+page_title: Upgrade An Existing Cluster to CRDs
+sidebar_title: Upgrade An Existing Cluster to CRDs
+description: >-
+  Upgrade an existing cluster to use custom resources.
+---
+
+# Upgrade An Existing Cluster to CRDs
+
+-> This feature requires consul-helm >= 0.26.0 and consul >= 1.8.4.
+
+If you have an existing Consul cluster running on Kubernetes you may need to perform
+extra steps to migrate to CRDs.
+
+You will need to perform extra steps if you are using:
+
+- Helm config `connectInject.centralConfig.defaultProtocol`, e.g.
+
+  ```yaml
+  connectInject:
+    centralConfig:
+      defaultProtocol: 'http'
+  ```
+
+- Setting the `consul.hashicorp.com/connect-service-protocol` annotation on your
+  connect pods, e.g.
+
+  ```yaml
+  annotations:
+    'consul.hashicorp.com/connect-service-protocol': 'http'
+  ```
+
+- Helm config `connectInject.centralConfig.proxyDefaults`, e.g.
+  ```yaml
+  connectInject:
+    centralConfig:
+      proxyDefaults: |
+        {
+          "local_connect_timeout_ms": 1000
+        }
+  ```
+
+## Migration Overview
+
+Overall our migration process will consist of identifying which [config entries](/docs/agent/config-entries)
+have been created in Consul and adding metadata to them so that they can
+be managed by a custom resource instead.
+
+## Why Migrate
```

Comment:
I'm wondering if this section should come before the first sentence? :idk:

Reply:
Do you mean before the Migration Overview section? That makes sense to me.